homeassistant/components/panasonic_viera/media_player.py

```diff
@@ -168,79 +168,136 @@ def supported_features(self):
             return SUPPORT_VIERATV | SUPPORT_TURN_ON
         return SUPPORT_VIERATV
 
+    @property
+    def device_info(self):
+        """Return device specific attributes."""
+        return {
+            "name": self.name,
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "manufacturer": DEVICE_MANUFACTURER,
+        }
+
+    def update(self):
+        """Retrieve the latest data and renew connection."""
+        if self._remote:
+            try:
+                self._muted = self._remote.get_mute()
+                self._volume = self._remote.get_volume() / 100
+                self._state = STATE_ON
+            except OSError:
+                self._state = STATE_OFF
+
+        old_remote = self._remote
```

Comment:
Should this block below be run only on error? It seems weird to do this always.

Reply:
It's because, at least with my TV, the service connection is always hanging. So I think it's great to 'renew' it every time the entity is updated so it never stops working. Would it be considered a bad practice?

Reply:
Is there no error if the connection is hanging? It would be better to only renew the connection on error.

Reply:
Yes, that makes sense.